### ISSUE TITLE
when I use calibre wireless to transfer the books, my kobo slept, then the connection closed.

### ISSUE THREAD

**Issue by mmschzs** [koreader/koreader#12611]
* KOReader version:newest
* Device:
kobo glo
#### Issue
when I use calibre wireless to transfer the books, my kobo slept, then the connection closed.
##### `crash.log` (if applicable)
`crash.log` is a file that is automatically created when KOReader crashes. It can normally be found in the KOReader directory:

* `/mnt/private/koreader` for Cervantes
* `koreader/` directory for Kindle
* `.adds/koreader/` directory for Kobo
* `applications/koreader/` directory for Pocketbook

Android logs are kept in memory. Please go to [Menu] → Help → Bug Report to save these logs to a file.

Please try to include the relevant sections in your issue description.
You can upload the whole `crash.log` file (zipped if necessary) on GitHub by dragging and dropping it onto this textbox.

If your issue doesn't directly concern a Lua crash, we'll quite likely need you to reproduce the issue with *verbose* debug logging enabled before providing the logs to us.
To do so, go to `Top menu → Hamburger menu → Help → Report a bug` and tap `Enable verbose logging`. Restart as requested, then repeat the steps for your issue.

If you instead opt to inline it, please do so behind a spoiler tag:
<details>
  <summary>crash.log</summary>

```
<Paste crash.log content here>
```
</details>

**Comment by pazos** [comment]
Yup, there's no mechanism to prevent that other than actively use the device while the books are being transferred or disable autostandby (or tweak its value) where you're about to send a bunch of books.

I don't know if autostandby has a way to be told it shouldn't do its duty but I'm not convinced it should be locked while a session with calibre is ongoing.

**Comment by NiLuJe** [comment]
Yeah, there was an implicit mechanism in place (via an `InputEvent`) sent in `UIManager:processZMQs`, but IIRC, we realiezd during the HTTP Inspector that it was annoying for that use-case, so we got rid of that, and what's left ion that function that appears to still do that is in fact a branch that will actually never be taken ;).

(Or... something, a bit fuzzy on the details rn ;p).

(We do have an explicit method to disable *standby*, but *suspend*, I don't think so?)

**Comment by pazos** [comment]
@NiLuJe: I have no idea about the standby thing. Please enlight me :)

Is there a way to reset the timeout or to increase n seconds its value? Because mostly everything **could** trigger this if the network is slow enough or the files long enough: wallabag? dictionary downloads? wikipedia to epub? ...

I really don't know how devices that use these potentially long network tasks handle autostandby. 

In this case it would be enough to increment the timeout on 5-10s for each document, given the timeout is capped to the max. Otherwise reset the timeout on each book transfer sound sensible.

**Comment by NiLuJe** [comment]
Basically, the PM plugin ensures we don't go to standby/suspend as long as `InputEvent` Events come in. That happens, automatically for, well, actual input events, of course; and processZMQs used to do it fairly aggressively, too for active ZMQ loops. IIRC, that turned out to be too aggressive for HttpInspector (or... something, can't recall), so it was moved to a branch that's unreachable because no ZMQ handler actually return anything..

There's also UIManager's preventStandby/allowStandby brackets for explicit non-interactive use-cases, but, IIRC, there's no equivalent for suspend.

**Comment by NiLuJe** [comment]
c.f., https://github.com/koreader/koreader/pull/11457/files/ff9dc5f47297b7c6551bd9ee984c00efbecb0bf2#diff-62aa9323c6e31b094e2ac64692de6fcaae44755c02d7f0a305a716bc5fafdbbc

**Comment by pazos** [comment]
Thank you!

So If I understand correctly:

1.  `AutoSuspend:preventStandby()`  
2. transfer books
3. `AutoSuspend:allowStandby()` 

Something tells me it couldn't be that simple..

**Comment by NiLuJe** [comment]
That's for standby only, we currently don't have anything of the sort for suspend.

We might want to do something similar for that, and just fold the bracketing inside insertZMQ & removeZMQ.

**Comment by ummm288** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.
